virtualenvwrapper.sh

```diff
@@ -759,16 +759,14 @@ function workon {
     # before switching so we use our override function,
     # if it exists, but make sure it's the deactivate function
     # we set up
-    type deactivate >/dev/null 2>&1
-    if [ $? -eq 0 ]
-    then
+    ! type deactivate >/dev/null 2>&1 || {
```

Comment:
Does this `{` start a sub shell?

Reply:
If it's a sub shell, then actions in the braces don't affect the current shell. Running and then unsetting deactivate won't actually do anything.

https://www.gnu.org/software/bash/manual/html_node/Command-Grouping.html#Command-Grouping seems to indicate that `{}` does _not_ create a new subshell.

Reply:
Oh okay! I thought it may have been a compatibility issue, some niche situation or even just bad practice.

I've been daily driving this change, so I know it at least fundamentally works (at least in my env)